src/server/game/Maps/Map.cpp

```diff
@@ -879,10 +879,15 @@ void Map::RemovePlayerFromMap(Player* player, bool remove)
 {
     sScriptMgr->OnPlayerLeaveMap(this, player);
 
+    player->getHostileRefManager().deleteReferences(); // multithreading crashfix
```

Comment:
References are two sided, if unit clears it's hostile reference, hostile unit removes it's threat reference. The rest is for the ai to handle.